api/handler/component_test.go

```diff
@@ -218,7 +207,45 @@ func (suite *ComponentTestSuite) TestCreateComponentWithReUsingPVCAsVolume() {
 	suite.Equal(reqComp.Volumes[0].PVToMatch, comp.Spec.Volumes[0].PVToMatch)
 }
 
-func (suite *ComponentTestSuite) TestCreateComponentWithReusePVInvalid() {
-	// todo
-	// 1. pv try to reuse is not available or not exist
+//func (suite *ComponentTestSuite) TestCreateComponentWithReusePVInvalid() {
+// todo
+// 1. pv try to reuse is not available or not exist
+//}
+
+func (suite *ComponentTestSuite) TestCreateComponentWithResourceRequirements() {
+	component := resources.Component{
+		Name: "component-with-resource-requirements",
+		ComponentSpec: v1alpha1.ComponentSpec{
+			Image: "foo",
+			ResourceRequirements: &coreV1.ResourceRequirements{
+				Limits: map[coreV1.ResourceName]resource.Quantity{
+					coreV1.ResourceCPU:    resource.MustParse("100m"),
+					coreV1.ResourceMemory: resource.MustParse("0.1Gi"),
+				},
+				Requests: map[coreV1.ResourceName]resource.Quantity{
+					coreV1.ResourceCPU:    resource.MustParse("10"),
+					coreV1.ResourceMemory: resource.MustParse("10Mi"),
+				},
+			},
```

Comment:
@wlchn update Component, 2 fields removed: `cpu` & `memory`, add `resourceRequirements`

ref:

Reply:
@mingmin-ddex I have a question here, How does create component's JSON structure like after change? Is that way below correct? 

before:
```
post url {
   name: "componentName1",
   cpu: "1Core",
   memory: "200Mi",
   ...otherFields
}
``` 
after(Is that correct?): 
```
post url {
   name: "componentName1",
   componentSpec: {
      resourceRequirements: {
          limits: {
            cpu: "1Core",
            memory: "200Mi",
          },
          requests: {
            cpu: "1Core",
            memory: "200Mi",
          }
 
      }
   }
   ...otherFields
}
```

Reply:
request example:

```json
{
    "name": "component-with-resource-requirements",
    "image": "foo",
    ...
    "resourceRequirements": {
        "limits": {
            "cpu": "100m",
            "memory": "107374182400m"
        },
        "requests": {
            "cpu": "10",
            "memory": "10Mi"
        }
    }
}
```